src/tableExplorer/tableExplorerWebViewController.ts

```diff
@@ -401,17 +401,18 @@ export class TableExplorerWebViewController extends ReactWebviewPanelController<
 
                 // Update result set with new row
                 if (state.resultSet) {
+                    // Create a completely new resultSet object to ensure React detects the change
                     state.resultSet = {
                         ...state.resultSet,
                         subset: [...state.resultSet.subset, result.row],
                         rowCount: state.resultSet.rowCount + 1,
                     };
 
-                    this.updateState();
-
                     this.logger.info(
                         `Added new row to result set, now has ${state.resultSet.rowCount} rows (${state.newRows.length} new)`,
                     );
+
+                    this.updateState();
```

Comment:
RegisterReducer already updates the state, so there’s no need to handle it explicitly unless you have a specific fix in mind. You can simply return the state.